src/components/map/MapContext.vue

```diff
@@ -0,0 +1,52 @@
+<script setup lang="ts">
+import { onMounted, ref } from 'vue'
+import initMap from '../../services/vcmap'
+
+onMounted(() => {
+  initMap(false, { x: -1.67, y: 48.1147, z: 16000, pitch: -10 }, 'map')
+})
+
+const is3d = ref(true)
+
+const toggle3d = () => {
+  initMap(is3d.value, { x: -1.67, y: 48.1147, z: 16000, pitch: -10 }, 'map')
```

Comment:
you should not init the map again and again while toggling 3D mode, the `mapCollection` is used for that with the `setActiveMap` method